### PR TITLE
aligned PST's for CSA Profiles

### PR DESCRIPTION
**Pull request by benrejebmoh**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.powsybl.openrao.data.craccreation.creator.csaprofile.craccreator;
```
Comment (bqth29) [inline data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java:1]: Add copyrights

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.powsybl.openrao.data.craccreation.creator.api.ImportStatus;
+import com.powsybl.openrao.data.craccreation.util.OpenRaoImportException;
+
+public class NotEnabledException extends OpenRaoImportException {
```
Comment (bqth29) [inline data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java:6]: Add author

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/PstRangeActionCreationTest.java [hunk 3]
```diff
@@ -153,4 +155,25 @@ void importPstSps60seconds() {
...
+
+    @Test
+    void checkAlignedPstsImport() {
+        CsaProfileCracCreationContext cracCreationContext = getCsaCracCreationContext("/profiles/remedialactions/AlignedPSTs.zip", NETWORK);
+        List<PstRangeAction> importedPstRangeActions = cracCreationContext.getCrac().getPstRangeActions().stream().sorted(Comparator.comparing(PstRangeAction::getId)).toList();
+        assertEquals(4, importedPstRangeActions.size());
+
+        assertPstRangeActionImported(importedPstRangeActions.get(0), "pst-range-action-4", "PST 4", "FFR2AA1  FFR4AA1  1", null, null, null);
+        assertHasOnInstantUsageRule(cracCreationContext, "pst-range-action-4", PREVENTIVE_INSTANT_ID, UsageMethod.AVAILABLE);
+        assertPstRangeActionImported(importedPstRangeActions.get(1), "tap-position-action-1-1", "RTE-tap-position-action-1-1", "BBE2AA1  BBE3AA1  1", null, null, "RTE");
```
Comment (bqth29) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/PstRangeActionCreationTest.java:167]: Can you also check the group of the PST RA and its speed?

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/PstRangeActionCreationTest.java [hunk 4]
```diff
@@ -153,4 +155,25 @@ void importPstSps60seconds() {
...
+        assertHasOnInstantUsageRule(cracCreationContext, "tap-position-action-1-1", PREVENTIVE_INSTANT_ID, UsageMethod.AVAILABLE);
+        assertPstRangeActionImported(importedPstRangeActions.get(2), "tap-position-action-1-2", "RTE-tap-position-action-1-2", "FFR2AA1  FFR4AA1  1", null, null, "RTE");
```
Comment (bqth29) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/PstRangeActionCreationTest.java:169]: Same

File: data/crac-creation/crac-creator-csa-profiles/src/test/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/PstRangeActionCreationTest.java [hunk 5]
```diff
@@ -153,4 +155,25 @@ void importPstSps60seconds() {
...
+        assertHasOnInstantUsageRule(cracCreationContext, "tap-position-action-1-2", PREVENTIVE_INSTANT_ID, UsageMethod.AVAILABLE);
+        assertPstRangeActionImported(importedPstRangeActions.get(3), "tap-position-action-2-1", "tap-position-action-2-1", "BBE2AA1  BBE3AA1  1", null, null, null);
```
Comment (bqth29) [inline data/crac-creation/crac-creator-csa-profiles/src/test/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/remedialaction/PstRangeActionCreationTest.java:171]: Same

File: data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import com.powsybl.openrao.data.craccreation.util.OpenRaoImportException;
+
+/**
+ * @author Mohamed Ben-rejeb {@literal <mohamed.ben-rejeb at rte-france.com>}
+ */
+public class NotEnabledException extends OpenRaoImportException {
+
+    public NotEnabledException(ImportStatus importStatus, String msg) {
+        super(importStatus, msg);
+    }
```
Comment (bqth29) [inline data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java:19]: I understand the idea of this exception class, but it appears OpenRaoImportException has no child class for all the other native CRAC formats.

@phiedw @pet-mit what do you think about this? Should we keep OpenRaoImportException or can we use this new class?

Comment (bqth29) [reply data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java:19]: Imo, keeping NotEnabledException would require a future refactoring to create specific classes for "common" import exceptions. To discuss.

Comment (phiedw) [reply data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java:19]: I agree with thomas, I think for now an OpenRaoImportException is probably better. Maybe we should open an issue to one day change all the exceptions to more specific ones? (including in the optimization part, not just the import)

Comment (benrejebmoh) [reply data/crac-creation/crac-creator-csa-profiles/src/main/java/com/powsybl/openrao/data/craccreation/creator/csaprofile/craccreator/NotEnabledException.java:19]: I also agree with you and wasn't convinced creating a specific exception especially like Thomas said OpenRaoImportException has no child class.
but OpenRaoImportException can't solve my problem on this code design. I want to create different behaviours inside PstRangeActionCreator depending on the use case: 
Allow a group having a pst not enabeled to be imported (ignoring only that pst)
Not allow the import of any pst of a group having a single not adequate pst (for example a pst having more than ONE StaticPropertyRange with direction RelativeDirectionKind.up) .
To make that possible i had to create a specific exception but of course we can discuss other possibilities

**Review by bqth29 — APPROVED** [review]

**Pull request merged**